src/main/java/com/github/manolo8/darkbot/backpage/HangaManager.java

```diff
@@ -0,0 +1,126 @@
+package com.github.manolo8.darkbot.backpage;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.backpage.hangar.Hangar;
+import com.github.manolo8.darkbot.backpage.hangar.HangarResponse;
+import com.github.manolo8.darkbot.core.itf.Tickable;
+import com.github.manolo8.darkbot.utils.Base64Utils;
+import com.github.manolo8.darkbot.utils.Time;
+import com.github.manolo8.darkbot.utils.http.Method;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class HangaManager implements Tickable {
+    private final Gson gson = new Gson();
+
+    private final Main main;
+    private final BackpageManager backpage;
+
+    private HangarResponse hangarList;
+    private HangarResponse currentHangar;
+
+    private long updateHangarListEvery = -1, updateCurrentHangarEvery = -1;
+    private long hangarListUpdatedUntil, currentHangarUpdatedUntil;
+
+
+    public HangaManager(Main main, BackpageManager backpage) {
+        this.main = main;
+        this.backpage = backpage;
+    }
+
+    @Override
+    public void tick() {
+        try {
+            if (updateHangarListEvery != -1 && hangarListUpdatedUntil < System.currentTimeMillis()) {
+                long timer = System.currentTimeMillis();
+                updateHangarList();
+                hangarListUpdatedUntil = System.currentTimeMillis() + updateHangarListEvery;
+
+                System.out.println("HangarList updated in: " + (System.currentTimeMillis() - timer) + "ms");
+                Time.sleep(500);
+            }
+
+            if (updateCurrentHangarEvery != -1 && currentHangarUpdatedUntil < System.currentTimeMillis()) {
+                long timer = System.currentTimeMillis();
+                updateCurrentHangar();
+                currentHangarUpdatedUntil = System.currentTimeMillis() + updateCurrentHangarEvery;
+
+                System.out.println("CurrentHangar updated in: " + (System.currentTimeMillis() - timer) + "ms");
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public HangarResponse getHangarList() {
+        return hangarList;
+    }
+
+    public HangarResponse getCurrentHangar() {
+        return currentHangar;
+    }
+
+    /**
+     * @param millis set -1 to disable
+     */
+    public void setUpdateCurrentHangarEvery(long millis) {
+        this.updateCurrentHangarEvery = millis;
+    }
```

Comment:
If it's intended for plugins to set this, that's an oversight. I'd rather have them keep on asking for the new data than automatically re-fetch it. If the plugin doesn't un-set this on uninstall, or throws an exception on install after having set this, it could lave the bot in a broken state (where it's re-fetching all the time) easily